model/project_parser_test.go

```diff
@@ -1227,57 +1226,99 @@ functions:
       env:
         CLIENT_URL: https://s3.amazonaws.com/mciuploads/evergreen/${task_id}/evergreen-ci/evergreen/clients/${goos}_${goarch}/evergreen
 `
+	paramsYAML := `
+  - command: myCommand
+    params:
+      env:
+        ${MY_KEY}: my-value
+        ${MY_NUMS}:
+        - 1
+        - 2
+        - 3`
 
 	for name, test := range map[string]func(t *testing.T){
 		"simpleYaml": func(t *testing.T) {
-			assert.NoError(t, checkProjectPersists([]byte(simpleYml)))
+			pp, errs := createIntermediateProject([]byte(simpleYml))
+			yamlToCompare, err := yaml.Marshal(pp)
+			assert.NoError(t, err)
+
+			assert.Len(t, errs, 0)
+			v := Version{
+				Id:            "my-version",
+				ParserProject: pp,
+				Config:        string(yamlToCompare),
+			}
+			assert.NoError(t, v.Insert())
+			newV, err := VersionFindOneId(v.Id)
+			assert.NoError(t, err)
+			require.NotNil(t, newV)
+			require.NotNil(t, newV.ParserProject)
+			require.Len(t, newV.ParserProject.Tasks, 1)
+			require.Len(t, newV.ParserProject.Tasks[0].Commands, 1)
+			assert.NotEmpty(t, newV.ParserProject.Tasks[0].Commands[0].Params)
+			assert.NotEmpty(t, newV.ParserProject.Tasks[0].Commands[0].ParamsYAML)
+
+			params := map[string]interface{}{}
+			assert.NoError(t, yaml.Unmarshal([]byte(newV.ParserProject.Tasks[0].Commands[0].ParamsYAML), &params))
+			val := params["env"]
+			env, ok := val.(map[interface{}]interface{})
+			assert.True(t, ok)
+			assert.Len(t, env, 2)
+			val1 := env["${MY_KEY}"]
+			val2 := env["${MY_NUMS}"]
+			assert.Equal(t, "my-value", fmt.Sprintf("%s", val1))
+			assert.Len(t, val2, 3)
+			assert.True(t, bytes.Contains([]byte(newV.Config), []byte(paramsYAML)))
 		},
 		"self-tests.yml": func(t *testing.T) {
 			filepath := filepath.Join(testutil.GetDirectoryOfFile(), "..", "self-tests.yml")
 			yml, err := ioutil.ReadFile(filepath)
 			assert.NoError(t, err)
-			assert.NoError(t, checkProjectPersists(yml))
+
+			pp, errs := createIntermediateProject([]byte(yml))
```

Comment:
If this is identical to the code in the "simpleYaml" block, you should be able to avoid this duplication.